src/Extensions.Hosting.AsyncInitialization/DependencyInjection/AsyncInitializationServiceCollectionExtensions.cs

```diff
@@ -49,10 +52,9 @@ public static IServiceCollection AddAsyncInitializer<TInitializer>(this IService
         public static IServiceCollection AddAsyncInitializer<TInitializer>(this IServiceCollection services, TInitializer initializer)
             where TInitializer : class, IAsyncInitializer
         {
-            if (initializer == null)
-                throw new ArgumentNullException(nameof(initializer));
-
-            return services
+            return initializer == null
+                ? throw new ArgumentNullException(nameof(initializer))
+                : services
```

Comment:
Please undo this change, it makes the code less readable.
(and it's not related to the feature you're implementing anyway)